src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -805,12 +811,70 @@ trait Implicits {
      *                             shadowing. The head of the list `iss` must represent implicits from the closest
      *                             enclosing scope, and so on.
      */
-    class ImplicitComputation(iss: Infoss, isLocalToCallsite: Boolean) {
+    class ImplicitComputation(iss: Infoss, isLocalToCallsite: Boolean) extends AbstractImplicitComputation(iss, isLocalToCallsite) {
+      /** Sorted list of eligible implicits.
+       */
+      val eligible: List[ImplicitInfo]  = {
+        val matches = iss flatMap { is =>
+          val result = is filter (info => checkValid(info.sym) && survives(info))
+          shadower addInfos is
+          result
+        }
+
+        // most frequent one first
+        matches sortBy (x => if (isView) -x.useCountView else -x.useCountArg)
+      }
+      if (eligible.nonEmpty)
+        printTyping(tree, eligible.size + s" eligible for pt=$pt at ${fullSiteString(context)}")
+    }
+
+    // MS: The split between the original and the new inductive computation could be more
+    // cleanly factored, but deferred to make it easier to see the difference between the
+    // cases.
```

Comment:
I think it would be beneficial if there was a step-by-step example in comments explaining inductive implicit lookup as per newly implemented logic. The test suite is extensive and even has approximate performance numbers (!), but personally for me a detailed example would be useful too.